src/instruction/instructions/add/adds.ts

```diff
@@ -24,17 +24,27 @@ export class AddsRegistersInstruction extends BaseInstruction {
   private rdPattern: string = '0001100000000XXX'
   private rnPattern: string = '0001100000XXX000'
   private rmPattern: string = '0001100XXX000000'
+  private expectedOptionCountMin: number = 2
+  private expectedOptionCountMax: number = 3
 
   public canEncodeInstruction(name: string, options: string[]): boolean {
     return (
       super.canEncodeInstruction(name, options) &&
-      isOptionCountValid(options, 2, 3) &&
+      isOptionCountValid(
+        options,
+        this.expectedOptionCountMin,
+        this.expectedOptionCountMax
+      ) &&
       options.every((x) => !isImmediate(x))
     )
   }
 
   public encodeInstruction(options: string[]): Halfword[] {
-    checkOptionCount(options, 2, 3)
+    checkOptionCount(
```

Comment:
Is a check like this still needed here? Because as soon as we are in here, the option count was already checked by the `canEncodeInstruction()`.

Reply:
You are not required to call canEncode method prior to encodeInstruction. Except from tests in our project in the instructionSet there is always first canEncode, but with future changes this may no longer be the case.